### PR TITLE
Generate image paths from mask paths, and allow no category mapping

### PR DESCRIPTION
**Pull request by jnothman**

### CODE REVIEW

File: weedcoco/importers/mask.py [hunk 1]
```diff
@@ -127,11 +113,64 @@ def _image_name_to_mask(name):
...
+def masks_to_coco(
+    path_pairs: Iterable[Tuple[str, str]],
+    color_to_category_map: Mapping[str, str],
+    check_consistent_dimensions: bool = True,
+):
+    """Converts images and masks to MS COCO images and annotations
+
+    Parameters
+    ----------
+    path_pairs : Iterable
```
Comment (ElevnLi) [inline weedcoco/importers/mask.py:145]: In this case the api of this module will be main function? or it's not the final version. I was expecting the arguments with only mask_dir as required and image_ext as optional

File: weedcoco/importers/mask.py [hunk 2]
```diff
@@ -127,11 +113,64 @@ def _image_name_to_mask(name):
...
+        image_file_name = f"{mask_path.stem}.{image_ext}"
+        yield mask_path, image_file_name
+
+    if not all_paths:
+        raise ValueError(f"No masks found in {mask_dir}")
+
+
+def masks_to_coco(
+    path_pairs: Iterable[Tuple[str, str]],
+    color_to_category_map: Mapping[str, str],
```
Comment (ElevnLi) [inline weedcoco/importers/mask.py:138]: Do we take category_color mapping as an arg, or we add an image processing step to extract all the colors and generate a mapping file?

Comment (jnothman) [reply weedcoco/importers/mask.py:138]: #467 has this functionality. I'll pull that branch into here for simplicity.

File: weedcoco/importers/mask.py (hunk 1 above)
Comment (jnothman) [reply weedcoco/importers/mask.py:145]: You're welcome to use it as a command-line tool, or by using this function and passing in the iterable from `generate_paths_from_mask_only`

**Review by ElevnLi — APPROVED** [review]
LGTM

**Pull request merged**